### PR TITLE
Residual Current Device [WIP]

### PR DESCRIPTION
**Pull request by TrojanCoyote**
## About The Pull Request

Adds a proc to shut off the APC that machinery is drawing from when someone gets the shockies 

## Why It's Good For The Game

It's somewhat realistic in how breakers/ devices work. Something causes a surge of electricity/ too much draw and the breaker flips to prevent further damage to equipment/people.

## Changelog

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
/:cl:

### CODE REVIEW

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -365,6 +365,9 @@
 		if(temp_area)
 			var/obj/machinery/power/apc/temp_apc = temp_area.get_apc()
 
+			if(temp_apc && temp_apc.terminal && temp_apc.terminal.powernet)
+				temp_apc.toggle_breaker()
```
Comment (TheLordME) [inline code/game/machinery/_machinery.dm:369]: DRY-Principle, why have two if-statements with the same conditions directly following one another? just move this into the other if

**Review by TheLordME — COMMENTED** [review]
WOOO

**Comment by Mazianni** [comment]
suggestion - this needs to trigger an alarm on the power network so engineers will eventually see it and fix it.

**Comment by TheLordME** [comment]
bump

**Comment by Mazianni** [comment]
I'll adopt this if Trojan would like.

**Comment by TheLordME** [comment]
> I'll adopt this if Trojan would like.

this whole thing is meant as a learning experience for @TrojanCoyote

**Pull request closed without merging**